programs/cleos/main.cpp

```diff
@@ -436,12 +431,52 @@ fc::variant push_transaction( signed_transaction& trx, const std::vector<public_
       if (tx_use_old_rpc) {
          return call(push_txn_func, packed_transaction_v0(trx, compression));
       } else {
-         try {
-            return call(send_txn_func, packed_transaction_v0(trx, compression));
-         }
-         catch (chain::missing_chain_api_plugin_exception &) {
-            std::cerr << "New RPC send_transaction may not be supported. Add flag --use-old-rpc to use old RPC push_transaction instead." << std::endl;
-            throw;
+         if( !amqp_address.empty() ) {
+            eosio::chain::named_thread_pool thread_pool( "amqp", 1 );
+            transaction_trace_msg trace_msg;
+            eosio::transaction_msg msg{packed_transaction( std::move( trx ), true, compression )};
+            auto buf = fc::raw::pack( msg );
+            const auto& tid = msg.get<packed_transaction>().id();
+            const string id = tid.str();
+            {
+               eosio::amqp qp_trx( fc::logger::get( DEFAULT_LOGGER ), thread_pool.get_executor(), amqp_address, "trx" );
+               eosio::amqp qp_trace( fc::logger::get( DEFAULT_LOGGER ), thread_pool.get_executor(), amqp_address,
+                                     "trace" );
+
+               qp_trx.publish( "", id, buf.data(), buf.size() );
+
+               std::promise<void> received_trace;
+               auto& consumer = qp_trace.consume();
+               consumer.onError( []( const char* message ) {
+                  std::cerr << "consume failed: " << message << std::endl;
+               } );
+               consumer.onReceived( [&]( const AMQP::Message& message, uint64_t delivery_tag, bool redelivered ) {
+                  if( message.hasCorrelationID() && message.correlationID() == id ) {
+                     fc::datastream<const char*> ds( message.body(), message.bodySize() );
+                     fc::raw::unpack( ds, trace_msg );
+                     qp_trace.ack( delivery_tag );
+                     received_trace.set_value();
+                  }
```

Comment:
Do we need to reject other messages?  What if there are multiple instances of cleos running in parallel?

Reply:
Probably safes to reject.

Reply:
I removed the ack as other consumers may be interested in the trace.